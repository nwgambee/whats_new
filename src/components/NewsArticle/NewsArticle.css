.NewsArticle {
  font-family: 'Raleway', sans-serif;
  border: 2px solid black;
  border-radius: 8px;
  width: 30%;
  margin: 20px;
  padding: 10px;
  background-color: #CEDADA;
  height: 640px;
}

img {
  width: 90%;
  height: 50%;
  border-radius: 3px;
}

.link {
  color: green;
  font-weight: bold;
  border: 1px solid black;
  border-radius: 3px;
  padding: 3px;
}

.description {
  margin-bottom: 10px;
}
