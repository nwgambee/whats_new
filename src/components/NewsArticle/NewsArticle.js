import React from 'react';
import './NewsArticle.css';

const NewsArticle = (props) => {
  return (
    <div className='NewsArticle'>
      <img src={props.img}/>
      <h2>{props.headline}</h2>
      <p className="description">{props.description}</p>
      <a className="link" href={props.url}>Read More</a>
    </div>
  )
}

export default NewsArticle;
