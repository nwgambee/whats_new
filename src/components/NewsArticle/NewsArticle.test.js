import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import NewsArticle from './NewsArticle';

describe('NewsArticle', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<NewsArticle
        img='some file'
        headline='Denver Man Eats Bug'
        description='A denver man has eaten a bug'
        url='some url'
          />);

    expect(wrapper).toMatchSnapshot();
  });
});
