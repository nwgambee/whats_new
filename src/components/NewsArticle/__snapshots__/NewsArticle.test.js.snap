// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsArticle should match the snapshot 1`] = `
<div
  className="NewsArticle"
>
  <img
    src="some file"
  />
  <h2>
    Denver Man Eats Bug
  </h2>
  <p
    className="description"
  >
    A denver man has eaten a bug
  </p>
  <a
    className="link"
    href="some url"
  >
    Article Link
  </a>
</div>
`;
